### ISSUE TITLE
CircleCI build-windows-vs2019 failure in Java test

### ISSUE THREAD

**Issue by akankshamahajan15** [facebook/rocksdb#12013]
CircleCI build-windows-vs2019 is failing with error-

   193>C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'C:\Users\circleci\project\build\CMakeFiles\6c28c13cc3f40b75e1d32e1c082fca9c\java_compiled_rocksdbjni_classes.rule;C:\Users\circleci\project\build\CMakeFiles\6c28c13cc3f40b75e1d32e1c082fca9c\java_class_filelist.rule;C:\Users\circleci\project\build\CMakeFiles\e3306321aeb3befa7242d5709768a54e\rocksdbjni_classes.jar.rule;C:\Users\circleci\project\build\CMakeFiles\d96c33f612259b7962a934bc8088c9c4\java_compiled_rocksdbjni_test_classes.rule;C:\Users\circleci\project\build\CMakeFiles\d96c33f612259b7962a934bc8088c9c4\java_class_filelist.rule;C:\Users\circleci\project\build\CMakeFiles\e3306321aeb3befa7242d5709768a54e\rocksdbjni_test_classes.jar.rule;C:\Users\circleci\project\build\CMakeFiles\d439f5a941803a37ff03cc9bef842ffc\rocksdbjni_test_classes.rule;C:\Users\circleci\project\java\CMakeLists.txt' exited with code 1. [C:\Users\circleci\project\build\java\rocksdbjni_test_classes.vcxproj]
   193>Done Building Project "C:\Users\circleci\project\build\java\rocksdbjni_test_classes.vcxproj" (default targets) -- FAILED.
   188>Done Building Project "C:\Users\circleci\project\build\java\rocksdbjni_test_classes.vcxproj.metaproj" (default targets) -- FAILED.

CircleCI result-
[https://app.circleci.com/pipelines/github/facebook/rocksdb/33769/workflows/5c5d05d9-ebf3-4b29-84c0-76818fd5f749/jobs/690144](https://app.circleci.com/pipelines/github/facebook/rocksdb/33769/workflows/5c5d05d9-ebf3-4b29-84c0-76818fd5f749/jobs/690144)

PR that might have introduced this - [#11756](https://github.com/facebook/rocksdb/pull/11756?fbclid=IwAR34QLZ3kqOzJ1HGcKzE1MgPPq4v4UeUP11o6aQBPP9N6FKWQTwxBde6PNs)

**Comment by akankshamahajan15** [comment]
@rhubner  Can you take a look at this one? Thanks
cc @adamretter

**Comment by rhubner** [comment]
@akankshamahajan15 It looks like it's failing already in build phase. Not the test one which I added. But I will try to look into it.

**Comment by rhubner** [comment]
I suspect little bit new JDK. Recently CircleCI changed default JDK from 20(non LTS) -> 21(LTS). Maybe It's related 🤔. Because nothing has changed between.

**Comment by adamretter** [comment]
@rhubner We should likely be enforcing JDK 8, as that is the baseline requirement for RocksJava

**Comment by rhubner** [comment]
@akankshamahajan15 I have quick fix to unblock build : #12014. 
Tomorrow I will discuss with @adamretter  How we should build RocksJava on Windows. Probably we need to download/install our version of JDK8.

**Comment by pdillinger** [comment]
It seems you're not heading toward the root cuase. The problem is definitely the changes to java/CMakeLists.txt in #11756. More detailed error output:

```
  193>CustomBuild:
         Generating CMakeFiles/rocksdbjni_classes.dir/java_class_filelist
         Creating Java archive rocksdbjni_classes.jar
   194>CustomBuild:
         Creating Java archive rocksdbjni_classes.jar
         java.nio.file.FileAlreadyExistsException: C:\Users\circleci\AppData\Local\Temp\rocksdbjni_classes.jar7913437267980843583.jar -> C:\Users\circleci\project\build\java\rocksdbjni_classes.jar
```

So there's an apparent race in the build where two things are trying to generate the same jar. If we look back here, we see a build with JDK21 without the changes and there's only one "Creating Java archive rocksdbjni_classes.jar": https://app.circleci.com/pipelines/github/facebook/rocksdb/33579/workflows/10a0fac3-3d74-4e3c-85b8-b41310ffd819/jobs/687565

Even a passing build with the changes attempts to build it twice: https://app.circleci.com/pipelines/github/facebook/rocksdb/33913/workflows/46fedeaf-4721-4fcc-8662-b5a803c368a3/jobs/693800

**Comment by rhubner** [comment]
@pdillinger You are right. It's trying to twice build `rocksdbjni_classes.jar`. 🤔

**Comment by ajkr** [comment]
Fixed by #12021
